### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by ryanwersal**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+jobs:
```
Comment (berdon) [inline azure-pipelines.yml:1]: I take it Azure Pipelines was dramatically better?

File: test/MorseL.Client.WebSockets.Tests/MorseL.Client.WebSockets.Tests.csproj [hunk 2]
```diff
@@ -5,18 +5,14 @@
...
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="xunit" Version="2.4.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\src\MorseL.Client.WebSockets\MorseL.Client.WebSockets.csproj" />
   </ItemGroup>
 
-  <ItemGroup>
```
Comment (berdon) [inline test/MorseL.Client.WebSockets.Tests/MorseL.Client.WebSockets.Tests.csproj:18]: Hm, I wonder what this was for?

File: test/MorseL.Client.WebSockets.Tests/WebSocketClientTests.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public async Task DisconnectingUnopenClientThrowsException()
             await Assert.ThrowsAsync<WebSocketClientException>(() => client.CloseAsync());
         }
 
-        [Fact]
+        [Fact(Skip = "This test fails on Linux/macOS but appears to succeed in Visual Studio's test runner?")]
```
Comment (berdon) [inline test/MorseL.Client.WebSockets.Tests/WebSocketClientTests.cs:37]: Likely a result of the underlying native web socket implementation. Can we just rig up the test to *pass* under *nix?

File: test/MorseL.Shared.Tests/MorseL.Shared.Tests.csproj [hunk 4]
```diff
@@ -10,10 +10,10 @@
     <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="2.0.0" />
```
Comment (berdon) [inline test/MorseL.Shared.Tests/MorseL.Shared.Tests.csproj:10]: We ought to introduce a solution level Directory.Build.Props for these versions:
https://github.com/berdon/Orleans.Providers.Redis/blob/master/Directory.Build.props

File: test/MorseL.Tests/EndToEndTests.cs [hunk 5]
```diff
@@ -191,7 +191,7 @@ public async void HubInvokingNonExistentClientMethodThrowsInHub(string methodNam
             }
         }
 
-        [Theory]
+        [Theory(Skip = "Flaky on Linux and macOS. Seems reliable on Windows.")]
```
Comment (berdon) [inline test/MorseL.Tests/EndToEndTests.cs:194]: Same as above? Treat it as expected to succeed under windows, expected to fail under *nix?

**Review by berdon — APPROVED** [review]
Comments

File: test/MorseL.Client.WebSockets.Tests/MorseL.Client.WebSockets.Tests.csproj (hunk 2 above)
Comment (ryanwersal) [reply test/MorseL.Client.WebSockets.Tests/MorseL.Client.WebSockets.Tests.csproj:18]: Not sure. Cleaning up the files and updating the dependencies resulted in removing some odd "multiple test adapters found for uri" messages and accurate detection of tests. If you look at the old test results you'll see a lot of shenanigans getting the tests to actually run before these modifications were made.

File: test/MorseL.Client.WebSockets.Tests/WebSocketClientTests.cs (hunk 3 above)
Comment (ryanwersal) [reply test/MorseL.Client.WebSockets.Tests/WebSocketClientTests.cs:37]: Yeah I'll definitely get these fixed with work on #48

File: test/MorseL.Shared.Tests/MorseL.Shared.Tests.csproj (hunk 4 above)
Comment (ryanwersal) [reply test/MorseL.Shared.Tests/MorseL.Shared.Tests.csproj:10]: Agreed. I made #51 to do that.

File: test/MorseL.Tests/EndToEndTests.cs (hunk 5 above)
Comment (ryanwersal) [reply test/MorseL.Tests/EndToEndTests.cs:194]: Regrettably it sometimes succeeds and sometimes fails on non-Windows platforms. For now, I'll leave it skipped and get that removed this weekend with work on #48

File: azure-pipelines.yml (hunk 1 above)
Comment (ryanwersal) [reply azure-pipelines.yml:1]: In my opinion: definitely. Particularly for .NET Core and getting it to run across many platforms.

**Pull request merged**